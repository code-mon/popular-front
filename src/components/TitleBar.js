import React from 'react';
import PropTypes from 'prop-types';
import { fadeInUp } from 'react-animations';
import Radium from 'radium';

const style = {
  base: {
    fontSize: '64px',
    fontWeight: 'bold',
    width: '1200px',
    margin: '0 auto',
    transform: 'translateY(11px)',
    animation: '1s ease-in',
    animationName: Radium.keyframes(fadeInUp, 'fadeInUp')
  }
};

const TitleCard = props => {
  return <div style={style.base}>{props.children}</div>;
};

TitleCard.propTypes = {
  children: PropTypes.node.isRequired
};

export default Radium(TitleCard);
