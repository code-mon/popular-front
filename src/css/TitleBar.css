.title-bar {
  animation: fadeInUp 2s linear;
}

@keyframes fadeInUp {
  0% {
    opacity: 0;
    transform: translateY(30px);
  }

  25% {
    opacity: 0.25;
    transform: translatY(20px);
  }

  50% {
    opacity: 0.5;
    transform: translateY(15px);
  }

  75% {
    opacity: 0.8;
    transform: translateY(11px);
  }

  100% {
    opactiry: 1;
    transform: translateY(7px);
  }
}
