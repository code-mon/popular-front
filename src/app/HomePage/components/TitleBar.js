import React from 'react';
import PropTypes from 'prop-types';
import '../../../css/TitleBar.css';

const style = {
  base: {
    fontSize: '64px',
    fontWeight: 'bold',
    width: '100%',
    maxWidth: '1200px',
    margin: '0 auto',
    transform: 'translateY(7px)'
  }
};

const TitleBar = props => {
  return (
    <div className="title-bar" style={style.base}>
      {props.children}
    </div>
  );
};

TitleBar.propTypes = {
  children: PropTypes.node.isRequired
};

export default TitleBar;
