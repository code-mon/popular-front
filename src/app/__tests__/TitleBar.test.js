import React from 'react';
import renderer from 'react-test-renderer';

import TitleBar from '../HomePage/components/TitleBar.js';

it('renders correctly', () => {
  const tree = renderer.create(<TitleBar>POPULAR</TitleBar>).toJSON();
  expect(tree).toMatchSnapshot();
});
