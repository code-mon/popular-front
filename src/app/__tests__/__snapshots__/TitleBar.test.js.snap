// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="title-bar"
  style={
    Object {
      "fontSize": "64px",
      "fontWeight": "bold",
      "margin": "0 auto",
      "maxWidth": "1200px",
      "transform": "translateY(7px)",
      "width": "100%",
    }
  }
>
  POPULAR
</div>
`;
